istio-pilot panic on invalid RbacConfig

**Describe the bug**
Invalid `RbacConfig` leads to `istio-pilot` nullref and crash loop.

**Expected behavior**
Validation prevents the invalid config.

**Steps to reproduce the bug**
This config (notice `exclusion` is null and not `{}`):
```
apiVersion: "rbac.istio.io/v1alpha1"
kind: RbacConfig
metadata:
  name: default
  namespace: istio-system
spec:
  mode: 'ON_WITH_EXCLUSION'
  exclusion:
```

Gets here (`li` is nil):
```
func isServiceInList(svc string, namespace string, li *rbacproto.RbacConfig_Target) bool {
	for _, ns := range li.Namespaces {
```

and crashes:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x1396096]

goroutine 386 [running]:
istio.io/istio/pilot/pkg/networking/plugin/authz.isServiceInList(0xc420a67660, 0x1c, 0xc420abcfc5, 0x5, 0x0, 0xc4200f4630)
        /workspace/go/src/istio.io/istio/pilot/pkg/networking/plugin/authz/rbac.go:298 +0x26
```

--- a/pilot/pkg/model/validation.go
+++ b/pilot/pkg/model/validation.go
@@ -1196,12 +1196,21 @@ func ValidateServiceRoleBinding(name, namespace string, msg proto.Message) error
 
 // ValidateRbacConfig checks that RbacConfig is well-formed.
 func ValidateRbacConfig(name, namespace string, msg proto.Message) error {
-	if _, ok := msg.(*rbac.RbacConfig); !ok {
+	in, ok := msg.(*rbac.RbacConfig)
+	if !ok {
 		return errors.New("cannot cast to RbacConfig")
 	}
 
 	if name != DefaultRbacConfigName {
 		return fmt.Errorf("rbacConfig has invalid name(%s), name must be %s", name, DefaultRbacConfigName)
+	}
+
+	if in.Mode == rbac.RbacConfig_ON_WITH_INCLUSION && in.Inclusion == nil {
+		return errors.New("inclusion cannot be null (use 'inclusion: {}' for none)")
+	}
+
+	if in.Mode == rbac.RbacConfig_ON_WITH_EXCLUSION && in.Exclusion == nil {
+		return errors.New("exclusion cannot be null (use 'exclusion: {}' for none)")
 	}
 
 	return nil

--- a/pilot/pkg/model/validation_test.go
+++ b/pilot/pkg/model/validation_test.go
@@ -2944,6 +2944,18 @@ func TestValidateRbacConfig(t *testing.T) {
 			name:     DefaultRbacConfigName,
 			in:       &rbac.RbacConfig{Mode: rbac.RbacConfig_ON},
 		},
+		{
+			caseName:     "empty exclusion",
+			name:         DefaultRbacConfigName,
+			in:           &rbac.RbacConfig{Mode: rbac.RbacConfig_ON_WITH_EXCLUSION},
+			expectErrMsg: "exclusion cannot be null (use 'exclusion: {}' for none)",
+		},
+		{
+			caseName:     "empty inclusion",
+			name:         DefaultRbacConfigName,
+			in:           &rbac.RbacConfig{Mode: rbac.RbacConfig_ON_WITH_INCLUSION},
+			expectErrMsg: "inclusion cannot be null (use 'inclusion: {}' for none)",
+		},
 	}
 	for _, c := range cases {
 		err := ValidateRbacConfig(c.name, c.namespace, c.in)

